test/dlc_tests/conftest.py

```diff
@@ -207,12 +207,23 @@ def generate_unique_values_for_fixtures(metafunc_obj, images_to_parametrize, val
             if key in metafunc_obj.fixturenames:
                 fixtures_parametrized[new_fixture_name] = []
                 for index, image in enumerate(images_to_parametrize):
+
+                    # Tag fixtures with EC2 instance types if env variable is present
+                    allowed_processors = ("gpu", "cpu", "eia")
+                    instance_tag = ""
+                    for processor in allowed_processors:
+                        if processor in image:
+                            instance_type = os.getenv(f"EC2_{processor.upper()}_INSTANCE_TYPE")
+                            if instance_type:
+                                instance_tag = f"-{instance_type.replace('.', '-')}"
+                                break
```

Comment:
It looks like this only inserts the instance type into the Name tag of the EC2 instance used for the test, and doesn't change the behavior of the actual test or the instance used for the test if there is no env variable of this name.

Is this understanding right?

Reply:
yep, that is correct.

Reply:
Oh sorry, misread your question. This should parametrize ec2_key_name with the instance type if the env variable is present.

Reply:
Yup. Thanks for stating it more precisely.